inference-engine/ie_bridges/java/org/intel/openvino/IECore.java

```diff
@@ -0,0 +1,95 @@
+package org.intel.openvino;
+
+import java.util.Map;
+
+public class IECore extends IEWrapper {
+    public static final String NATIVE_LIBRARY_NAME = "inference_engine_java_api";
+
+    public IECore() {
+        super(GetCore());
+    }
+
+    public IECore(String xmlConfigFile) {
+        super(GetCore_1(xmlConfigFile));
+    }
+
+    public CNNNetwork ReadNetwork(final String xml, final String bin) {
+        return new CNNNetwork(ReadNetwork1(nativeObj, xml, bin));
+    }
+
+    public CNNNetwork ReadNetwork(final String xml) {
+        return new CNNNetwork(ReadNetwork(nativeObj, xml));
+    }
+
+    public ExecutableNetwork LoadNetwork(CNNNetwork net, final String device) {
+        return new ExecutableNetwork(LoadNetwork(nativeObj, net.getNativeObjAddr(), device));
+    }
+
+    public ExecutableNetwork LoadNetwork(CNNNetwork net, final String device, final Map<String, String> config) {
+        return new ExecutableNetwork(LoadNetwork1(nativeObj, net.getNativeObjAddr(), device, config));
+    }
+
+    public void RegisterPlugin(String pluginName, String deviceName) {
+        RegisterPlugin(nativeObj, pluginName, deviceName);
+    }
+
+    public void RegisterPlugin(String xmlConfigFile) {
+        RegisterPlugins(nativeObj, xmlConfigFile);
+    }
+
+    public void UnregisterPlugin(String deviceName) {
+        UnregisterPlugin(nativeObj, deviceName);
+    }
+
+    public void AddExtension(String extension) {
+        AddExtension(nativeObj, extension);
+    }
+
+    public void AddExtension(String extension, String deviceName) {
```

Comment:
we can omit this method. `AddExtension` without device adds extensions to a common map and the devices use it.

Reply:
But in python bindings we have only ```add_extension``` method with device:
https://docs.openvinotoolkit.org/latest/ie_python_api/classie__api_1_1IECore.html#a8a4b671a9928c7c059bd1e76d2333967

Reply:
Yeah, we call native methods and there is a method which accepts device name as well: https://docs.openvinotoolkit.org/latest/classInferenceEngine_1_1Core.html#aa1ddb53f4160bf0735239f4aa0c12320. So it's easier for port from/to C++/Python.

Reply:
`AddExtension` with no device just puts extensions to internal list in `Core`. After, all the devices use it. So, we probably need to deprecated `AddExtension` with device in C++. So, no needs to expose it in Java.